cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -67,49 +68,71 @@ func DeployedDNSAddon(client clientset.Interface) (string, error) {
 	}
 }
 
-// deployedDNSReplicas returns the replica count for the current DNS deployment
-func deployedDNSReplicas(client clientset.Interface, replicas int32) (*int32, error) {
+// defaultResourceRequirements returns the default resourceRequirements
+func defaultResourceRequirements() *v1.ResourceRequirements {
+	return &v1.ResourceRequirements{
+		Limits: v1.ResourceList{
+			v1.ResourceMemory: apiresource.MustParse("170Mi"),
+		},
+		Requests: v1.ResourceList{
+			v1.ResourceCPU:    apiresource.MustParse("0.1"),
+			v1.ResourceMemory: apiresource.MustParse("70Mi"),
+		},
+	}
+}
+
+// deployedDNSDeployment returns the replica count and the resourceRequirements for the current DNS deployment
+func deployedDNSDeployment(client clientset.Interface, replicas int32) (*int32, *v1.ResourceRequirements, error) {
 	deploymentsClient := client.AppsV1().Deployments(metav1.NamespaceSystem)
 	deployments, err := deploymentsClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "k8s-app=kube-dns"})
+
+	defaultResourceRequirements := defaultResourceRequirements()
+
 	if err != nil {
-		return &replicas, errors.Wrap(err, "couldn't retrieve DNS addon deployments")
+		return &replicas, defaultResourceRequirements, errors.Wrap(err, "couldn't retrieve DNS addon deployments")
 	}
 	switch len(deployments.Items) {
 	case 0:
-		return &replicas, nil
+		return &replicas, defaultResourceRequirements, nil
 	case 1:
-		return deployments.Items[0].Spec.Replicas, nil
+		return deployments.Items[0].Spec.Replicas, &deployments.Items[0].Spec.Template.Spec.Containers[0].Resources, nil
 	default:
-		return &replicas, errors.Errorf("multiple DNS addon deployments found: %v", deployments.Items)
+		return &replicas, defaultResourceRequirements, errors.Errorf("multiple DNS addon deployments found: %v", deployments.Items)
 	}
 }
 
 // EnsureDNSAddon creates the CoreDNS addon
 func EnsureDNSAddon(cfg *kubeadmapi.ClusterConfiguration, client clientset.Interface, out io.Writer, printManifest bool) error {
 	var replicas *int32
+	var resourceRequirements *v1.ResourceRequirements
 	var err error
 	if !printManifest {
-		replicas, err = deployedDNSReplicas(client, coreDNSReplicas)
+		replicas, resourceRequirements, err = deployedDNSDeployment(client, coreDNSReplicas)
```

Comment:
can we just return a Deployment object and take what we need from it?

Reply:
should be possible :) Will update the PR tomorrow